src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/X500NameEncoder.cs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal static class X500NameEncoder
+    {
+        private static readonly char[] s_quoteNeedingChars =
+       {
+            ',',
+            '+',
+            '=',
+            '\"',
+            '\n',
+            // \r is NOT in this list.
+            '<',
+            '>',
+            '#',
+            ';',
+        };
+
+        internal static string X500DistinguishedNameDecode(
+            byte[] encodedDistinguishedName,
+            bool printOid,
+            X500DistinguishedNameFlags flags)
+        {
+            bool reverse = flags.HasFlag(X500DistinguishedNameFlags.Reversed);
+            bool quoteIfNeeded = !flags.HasFlag(X500DistinguishedNameFlags.DoNotUseQuotes);
+            string dnSeparator;
+
+            if (flags.HasFlag(X500DistinguishedNameFlags.UseSemicolons))
+            {
+                dnSeparator = "; ";
+            }
+            else if (flags.HasFlag(X500DistinguishedNameFlags.UseNewLines))
+            {
+                dnSeparator = Environment.NewLine;
+            }
+            else
+            {
+                // This is matching Windows (native) behavior, UseCommas does not need to be asserted,
+                // it is just what happens if neither UseSemicolons nor UseNewLines is specified.
+                dnSeparator = ", ";
+            }
+
+            using (SafeX509NameHandle x509Name = Interop.Crypto.DecodeX509Name(encodedDistinguishedName, encodedDistinguishedName.Length))
+            {
+                if (x509Name.IsInvalid)
+                {
+                    return "";
+                }
+
+                // GitHub.com's SSL cert has a "pretty long" subject (partially due to the unknown OIDs):
```

Comment:
Maybe clarify earlier on that this is an example being used for length approximation.  Wasn't clear to me why we were talking about GitHub's cert until the very end of the comment.